### ISSUE TITLE
Managing collections when calling all from Fortnox

### ISSUE THREAD

**Issue by ehannes** [accodeing/fortnox-api#25]
When fetching all entities from Fortnox, the response omits many details. For instance, calling https://api.fortnox.se/3/customers returns `Customer`s with 10 attributes. Calling https://api.fortnox.se/3/customers/1 returns a `Customer` with ~60 attributes.

**Comment by d-Pixie** [comment]
As we talked about on Slack I think we should manage this by:
 1. Creating a simple customer class, `Model::Simple::Customer`, to hold the attributes that are common to both a partial and full resource.
 2. Include the simple customer in the full customer, to prevent duplication.
 3. Add a `Collection`class that wraps `Array` and can act as the collection proxy for partial/full customers.
 4. Implement a `.full` method on both the `Simple::Customer` and `Collection` classes that returns a new instance of `Customer` and `Collection` respectively, but with the `Simple::Customer` instances in the collection replaced with the corresponding `Customer` instances instead.
 5. Document this and underline that it's VERY expensive to do `.full` on a large collection.

I have also done some reading on this and while it's not uncommon to allow for partial loads like this it's commonly done by the requesting party telling the server what info it needs. I thought I would look up some libraries that deal with REST APIs that support partial collections and see how they do it before I make a decision.

But if you have any comments in the meanwhile they are welcome :)

**Comment by d-Pixie** [comment]
I never found a good reference for such an implementation so I think we will go with the above plan for this.

**Comment by d-Pixie** [comment]
Found this now: https://github.com/BookingSync/ember-data-partial-model

They provide a method to explicitly load the full model, but not too fully load a collection. They also do implicit full model loading if you try to access a property that is not available on the partial model...

The circumstances are quite different though. When you do implicit loading they return null instantly and then load the model in the background. The ember data bindings will update the view, replacing the null value, as soon as the model had loaded. Meanwhile the original code just keeps running.

So while that looks like a really good solution for frontend JavaScript, it won't work that well for us. So I think we'll stick with only explicit loading of partial models and a convenience method on collection to trigger full loading of the entire collection.

We'll have to put some warnings in the docs though and try to get Fortnox to implement client side attribute selection for partial loading down the line.

**Comment by ehannes** [comment]
Yep. In my case, I know about this limitation and I can live with it. I think we can release a first alpha version with a warning in the docs. I changed milestone for this issue to "First public release".

**Comment by ehannes** [comment]
In version 1.0.0 you have access `meta.partial?` which tells you if the model is only partially loaded (via a collection). Let's consider that as good enough for now.
